### ISSUE TITLE
Tokenizer padding error

### ISSUE THREAD

**Issue by zhaochenyang20** [neulab/prompt2model#63]
Previously, we are using:

```py
if self.tokenizer.pad_token is None:
	self.tokenizer.add_special_tokens({"pad_token": "[PAD]"})
if self.model.config.pad_token_id is None:
	self.model.config.pad_token_id = self.model.config.eos_token_id
```

This will fail due to we set `pad_token` as `[PAD]`, but assign `pad_token_id` as `eos_token_id`.

We can change it this way:

```py
if self.tokenizer.pad_token is None:
	self.tokenizer.pad_token = self.tokenizer.eos_token
	self.model.config.pad_token_id = self.model.config.eos_token_id
	self.model.config.attention_mask_fn = lambda input_ids: (input_ids != self.model.config.pad_token_id).float()
```

Setting the `attention_mask_fn` is not necessary because `[EOS]` will automatically set the attention mask as 0.

The previous one may not be optimal:

> Using the EOS token as a padding token may not be ideal because it is semantically meaningful. A separate padding token, such as [PAD], is generally preferred to avoid potential confusion or interference with the model's understanding and generation process.

So my best answer is:

```py
if self.tokenizer.pad_token is None:
    self.tokenizer.add_special_tokens({"pad_token": "[PAD]"})
if self.model.config.pad_token_id is None:
    self.model.config.pad_token_id = len(self.tokenizer)
    self.model.resize_token_embeddings(len(self.tokenizer))
    self.model.config.attention_mask_fn = lambda input_ids: (
        input_ids != self.model.config.pad_token_id
    ).float()
```
